e2e/Tests/iothub/device/DeviceClientX509AuthenticationE2ETests.cs

```diff
@@ -27,82 +26,79 @@ public class DeviceClientX509AuthenticationE2ETests : E2EMsTestBase
-        public async Task X509_InvalidDeviceId_Throw_UnauthorizedException_AmqpWs()
-        {
-            await X509InvalidDeviceIdOpenAsyncTest(new IotHubClientAmqpSettings(IotHubClientTransportProtocol.WebSocket)).ConfigureAwait(false);
+            await X509InvalidDeviceIdOpenAsyncTest(new IotHubClientAmqpSettings(protocol), ct).ConfigureAwait(false);
         }
 
+        [DataTestMethod]
         // TODO: there is a problem with DotNetty on net6.0 with gatewayv2. Wait for transition to MqttNet to re-enable and try this test out.
 #if !NET6_0_OR_GREATER
-        [TestMethod]
-        [Timeout(TestTimeoutMilliseconds)]
-        public async Task X509_InvalidDeviceId_Throw_UnauthorizedException_MqttTcp()
-        {
-            await X509InvalidDeviceIdOpenAsyncTest(new IotHubClientMqttSettings()).ConfigureAwait(false);
-        }
+        [DataRow(IotHubClientTransportProtocol.Tcp)]
 #endif
-
-        [TestMethod]
-        [Timeout(TestTimeoutMilliseconds)]
-        public async Task X509_InvalidDeviceId_Throw_UnauthorizedException_MqttWs()
+        [DataRow(IotHubClientTransportProtocol.WebSocket)]
+        public async Task X509_InvalidDeviceId_Throw_UnauthorizedException_Mqtt(IotHubClientTransportProtocol protocol)
         {
-            await X509InvalidDeviceIdOpenAsyncTest(new IotHubClientMqttSettings(IotHubClientTransportProtocol.WebSocket)).ConfigureAwait(false);
-        }
+            // Seeting up one cancellation token for the complete test flow
+            using var cts = new CancellationTokenSource(s_testTimeout);
+            CancellationToken ct = cts.Token;
 
-        [TestMethod]
-        [Timeout(TestTimeoutMilliseconds)]
-        public async Task X509_Enable_CertificateRevocationCheck_MqttTcp()
-        {
-            IotHubClientTransportSettings transportSetting = CreateMqttTransportSettingWithCertificateRevocationCheck(IotHubClientTransportProtocol.Tcp);
-            await SendMessageTestAsync(transportSetting).ConfigureAwait(false);
+            await X509InvalidDeviceIdOpenAsyncTest(new IotHubClientMqttSettings(protocol), ct).ConfigureAwait(false);
         }
 
-        [TestMethod]
-        [Timeout(TestTimeoutMilliseconds)]
-        public async Task X509_Enable_CertificateRevocationCheck_MqttWs()
+        [DataTestMethod]
+        [DataRow(IotHubClientTransportProtocol.Tcp)]
+        [DataRow(IotHubClientTransportProtocol.WebSocket)]
+        public async Task X509_Enable_CertificateRevocationCheck_Mqtt(IotHubClientTransportProtocol protocol)
         {
-            IotHubClientTransportSettings transportSetting = CreateMqttTransportSettingWithCertificateRevocationCheck(IotHubClientTransportProtocol.WebSocket);
-            await SendMessageTestAsync(transportSetting).ConfigureAwait(false);
-        }
+            // Seeting up one cancellation token for the complete test flow
+            using var cts = new CancellationTokenSource(s_testTimeout);
+            CancellationToken ct = cts.Token;
```

Comment:
Is it really worth making an alias for `cts.Token` when it is only used once instead of passing it directly on line 65?

Reply:
I see your point. I'll leave it as it is for now though, we can refactor it in the future if we want.